foursight_core/app_utils.py

```diff
@@ -424,17 +457,34 @@ def auth0_callback(self, request, env):
         }
         json_payload = json.dumps(payload)
         headers = {'content-type': "application/json"}
-        res = requests.post(self.OAUTH_TOKEN_URL, data=json_payload, headers=headers)
+        if self.auth0_domain:
+            token_url = f'https://{self.auth0_domain}/oauth/token'
+        else:
+            token_url = self.OAUTH_TOKEN_URL
+        res = requests.post(token_url, data=json_payload, headers=headers)
         id_token = res.json().get('id_token', None)
+        expires_in = res.json().get('expires_in', None)
+
+        # store redis token if turned on
+        conn = self.init_connection(env)
+        redis_handle = conn.get_redis_base()
+        if redis_handle:
+            redis_session_token = RedisSessionToken(
+                namespace=env, jwt=id_token
+            )
+            redis_session_token.store_session_token(redis_handler=redis_handle)
+            # overwrite id_token in this case to be the session token
+            id_token = redis_session_token.get_session_token()
+            expires_in = (3 * 60 * 59)  # default session token expiration is 3 hours
+
         if id_token:
-            expires_in = res.json().get('expires_in', None)
             if domain and not self.is_running_locally(req_dict):
-                cookie_str = ''.join(['jwtToken=', id_token, '; Domain=', domain, '; Path=/;'])
+                cookie_str = ''.join(['c4_st=', id_token, '; Domain=', domain, '; Path=/;'])
```

Comment:
Untested:
```suggestion
                cookie_str = f'c4_st={id_token}; Domain={domain}; Path=/;'
```
This also highlights that you're assuming the domain is syntactically compatible. I guess there's not an injection worry here but code like this that doesn't escape data coming in always makes me nervous. It's _probably_ fine.